crates/validation/src/collection.rs

```diff
@@ -441,8 +460,13 @@ pub fn walk_selector(
 const FLOW_DOCUMENT: &str = "flow_document";
 /// The default field name for the document publication time.
 const FLOW_PUBLISHED_AT: &str = "flow_published_at";
+/// The default field name for the truncation sentinel.
+const FLOW_TRUNCATED: &str = "flow_truncated";
```

Comment:
this should be named `_meta/flow_truncated` -- the implicit projection of `/_meta/flow_truncated` -- as materializations use the field name to determine whether it's recommended or not.

Also, and this is half baked, but I suspect the validation logic might get easier to reason about if we _don't_ think of `/_meta/flow_truncated` as the projection pointer. Conceptually, this is a value that exists outside of the document, and simply can't be referenced via pointer.

Perhaps it's just tacking on a Projection like:

Projection {
  field: '_meta/flow_truncated',
  ptr: '',
  contentEncoding: 'flow_truncated'
}

after first verifying there isn't an implicit or explicit projection of that name, so we can provide a good error message.

Reply:
The field name is updated.

> Also, and this is half baked, but I suspect the validation logic might get easier to reason about if we don't think of /_meta/flow_truncated as the projection pointer. Conceptually, this is a value that exists outside of the document, and simply can't be referenced via pointer.

I don't think I see what this is buying us in terms of simplicity. I'm open to discussing more if there's something I'm missing. But the way I see it now it's that we can either match against a phony JSON pointer or a phony content encoding, and I see them as more or less equivalently distasteful. So I'm inclined to leave it as is.